lms/djangoapps/course_api/blocks/tests/test_views.py

```diff
@@ -491,3 +491,97 @@ def test_completion_all_course_with_nav_depth(self):
         })
         for block_id in self.non_orphaned_block_usage_keys:
             assert response.data['blocks'][block_id].get('completion')
+
+
+class TestBlockMetadataView(SharedModuleStoreTestCase):  # pylint: disable=test-inherits-tests
+    """
+    Test class for BlocksInCourseView
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # create a toy course
+        cls.course = ToyCourseFactory.create(
+            modulestore=cls.store,
+            due=datetime(3013, 9, 18, 11, 30, 00),
+        )
+        cls.course_key = cls.course.id
+        cls.course_usage_key = cls.store.make_course_usage_key(cls.course_key)
+
+        cls.non_orphaned_block_usage_keys = {
+            str(item.location)
+            for item in cls.store.get_items(cls.course_key)
+            # remove all orphaned items in the course, except for the root 'course' block
+            if cls.store.get_parent_location(item.location) or item.category == 'course'
+        }
+
+    def setUp(self):
+        super().setUp()
+        self.admin_user = AdminFactory.create()
+        self.client.login(username=self.admin_user.username, password='test')
+        self.usage_key = list(self.non_orphaned_block_usage_keys)[0]
+        self.url = reverse(
+            'blocks_metadata',
+            kwargs={'usage_key_string': str(self.usage_key)}
+        )
+        self.query_params = {'include': "index_dictionary"}
+
+    def verify_response(self, expected_status_code=200, params=None, url=None, cacheable=False):
+        """
+        Ensure that sending a GET request to the specified URL returns the
+        expected status code.
+
+        Arguments:
+            expected_status_code: The status_code that is expected in the
+                response.
+            params: Parameters to add to self.query_params to include in the
+                request.
+            url: The URL to send the GET request.  Default is self.url.
+
+        Returns:
+            response: The HttpResponse returned by the request
+        """
+        if params:
+            self.query_params.update(params)
+        response = self.client.get(url or self.url, self.query_params)
+        assert response.status_code == expected_status_code, str(response.content)
+        if cacheable:
+            assert response.get('Cache-Control', None) == 'max-age={}'.format(
+                settings.CACHE_MIDDLEWARE_SECONDS
+            )
+        else:
+            assert response.get('Cache-Control', None) is None
```

Comment:
The `cacheable` parameter doesn't seem to be used. So you should either use it in the `test_block_metadata_response` or delete the parameter + the six lines of code above.